connectd/peer_exchange_initmsg.c

```diff
@@ -177,6 +238,34 @@ struct io_plan *peer_exchange_initmsg(struct io_conn *conn,
 	tlvs->networks = tal_dup_arr(tlvs, struct bitcoin_blkid,
 				     &chainparams->genesis_blockhash, 1, 0);
 
+#if EXPERIMENTAL_FEATURES /* BOLT1 remote_addr #917 */
+	/* BOLT-remote-address #1:
+	 * The sending node:
+	 * ...
+	 *  - SHOULD set `remote_addr` to reflect the remote IP address (and port) of an
+	 *    incoming connection, if the node is the receiver and the connection was done
+	 *    via IP.
+	 */
+	if (incoming && addr->itype == ADDR_INTERNAL_WIREADDR) {
+		if (addr->u.wireaddr.type == ADDR_TYPE_IPV4) {
+			remote_addr_v4 = tal(tlvs, struct remote_addr_ipv4);
+			remote_addr_v4->type = 1;
+			remote_addr_v4->port = addr->u.wireaddr.port;
+			memcpy(&remote_addr_v4->addr,
+			       addr->u.wireaddr.addr, 4);
+			tlvs->remote_addr = (u8*)remote_addr_v4;
+		}
+		if (addr->u.wireaddr.type == ADDR_TYPE_IPV6) {
+			remote_addr_v6 = tal(tlvs, struct remote_addr_ipv6);
+			remote_addr_v6->type = 2;
+			remote_addr_v6->port = addr->u.wireaddr.port;
+			memcpy(&remote_addr_v6->addr,
+			       addr->u.wireaddr.addr, 16);
+			tlvs->remote_addr = (u8*)remote_addr_v6;
+		}
```

Comment:
is it better use the switch case here? and abort in case of not match?

From the spec https://github.com/lightning/bolts/pull/917/commits/efe9d2d151b80b1992f0ed65fead2b36d2908721 I understand that here can be possible only two type of values